Makefile

```diff
@@ -123,6 +123,8 @@ ref.c      := $(sort $(wildcard backends/ref/*.c))
 template.c := $(sort $(wildcard backends/template/*.c))
 cuda.c     := $(sort $(wildcard backends/cuda/*.c))
 cuda.cu    := $(sort $(wildcard backends/cuda/*.cu))
+cuda-reg.c := $(sort $(wildcard backends/cuda-reg/*.c))
```

Comment:
If these end up being maintained in the same folder, then you can have fewer changes here.